src/hats_import/verification/run_verification.py

```diff
@@ -1,14 +1,439 @@
+        print(f"\nStarting: {description}")
+
+        passed = hats.io.validation.is_valid_catalog(target, strict=True)
+        self.results.append(Result(test=test, description=description, passed=passed, target=target.name))
+        print(f"Result: {'PASSED' if passed else 'FAILED'}")
+        return passed
+
+    def test_file_sets(self) -> bool:
+        """Test that files in _metadata match the parquet files on disk. Add one `Result` to `results`.
+
+        This is a simple test that can be especially useful to run after copying or moving
+        the catalog to a different local or cloud-based destination.
+
+        Returns
+        -------
+            bool: True if the file sets match, else False.
+        """
+        # info for the report
+        description = "Test that files in _metadata match the data files on disk."
+        print(f"\nStarting: {description}")
+
+        _failed_files = set(self.files_ds.files).symmetric_difference(self.metadata_ds.files)
+        failed_files = [str(Path(file).relative_to(self.args.input_dataset_path)) for file in _failed_files]
+        passed = len(failed_files) == 0
+        self.results.append(
+            Result(
+                passed=passed,
+                test="file sets",
+                target="_metadata vs data files",
+                bad_files=failed_files,
+                description=description,
+            )
+        )
+
+        print(f"Result: {'PASSED' if passed else 'FAILED'}")
+        return passed
+
+    def test_num_rows(self) -> bool:
+        """Test the number of rows in the dataset. Add `Result`s to `results`.
+
+        File footers are compared with _metadata and the user-supplied truth (if provided).
+
+        Returns
+        -------
+        bool: True if all checks pass, else False.
+        """
+        test = "num rows"
+        description = "Test that number of rows are equal."
+        print(f"\nStarting: {description}")
+
+        # get the number of rows in each file, indexed by file path. we treat this as truth.
+        files_df = self._load_nrows(self.files_ds)
+        files_df_total = f"file footers ({files_df.num_rows.sum():,})"
+
+        # check _metadata
+        target = "file footers vs _metadata"
+        print(f"\t{target}")
+        metadata_df = self._load_nrows(self.metadata_ds)
+        row_diff = files_df - metadata_df
+        failed_frags = row_diff.loc[row_diff.num_rows != 0].index.to_list()
+        passed_md = len(failed_frags) == 0
+        _description = f" {files_df_total} vs _metadata ({metadata_df.num_rows.sum():,})."
+        self.results.append(
+            Result(
+                passed=passed_md,
+                test=test,
+                bad_files=failed_frags,
+                target=target,
+                description=description + _description,
+            )
+        )
+
+        # check user-supplied total, if provided
+        if self.args.truth_total_rows is not None:
```

Comment:
There is also a `catalog.catalog_info.total_rows` attribute that should match the user provided value. And that should match the `_metadata`-derived value.

Reply:
Added a test against the catalog info num rows.